include/libdnf/rpm/repo.hpp

```diff
@@ -118,262 +109,349 @@ struct Repo {
+    /// @param headers nullptr terminated array of C strings
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.setHttpHeaders(const char * headers[])
+    void set_http_headers(const char * headers[]);
+
+    /// Get array of added/changed/removed http headers.
+    /// @return nullptr terminated array of C strings
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.getHttpHeaders()
+    const char * const * get_http_headers() const;
+
+    /// Returns mirrorlist associated with the repository.
+    /// Mirrors on this list are mirrors parsed from mirrorlist/metalink specified by LRO_MIRRORLIST or
+    /// from mirrorlist specified by LRO_MIRROSLISTURL and metalink specified by LRO_METALINKURL.
+    /// No URLs specified by LRO_URLS are included in this list.
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.getMirrors()
+    std::vector<std::string> get_mirrors() const;
+
+    /// Sets substitutions. Substitutions are used to substitute variables in repository configuration.
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.setSubstitutions(const std::map<std::string, std::string> & substitutions)
+    void set_substitutions(const std::map<std::string, std::string> & substitutions);
 
     ~Repo();
 
     class Impl;
+
 private:
     friend struct PackageTarget;
-    friend Impl * repoGetImpl(Repo * repo);
-    std::unique_ptr<Impl> pImpl;
+    std::unique_ptr<Impl> p_impl;
 };
 
 struct Downloader {
 public:
-    static void downloadURL(ConfigMain * cfg, const char * url, int fd);
+    static void download_url(ConfigMain * cfg, const char * url, int fd);
 };
 
-/**
-* @class PackageTargetCB
-*
-* @brief Base class for PackageTarget callbacks
-*
-* User implements PackageTarget callbacks by inheriting this class and overriding its methods.
-*/
+/// Base class for PackageTarget callbacks
+/// User implements PackageTarget callbacks by inheriting this class and overriding its methods.
 class PackageTargetCB {
 public:
-    /** Transfer status codes */
-    enum class TransferStatus {
-        SUCCESSFUL,
-        ALREADYEXISTS,
-        ERROR
-    };
+    // Transfer status codes
+    enum class TransferStatus { SUCCESSFUL, ALREADYEXISTS, ERROR };
 
     virtual int end(TransferStatus status, const char * msg);
-    virtual int progress(double totalToDownload, double downloaded);
-    virtual int mirrorFailure(const char *msg, const char *url);
+    virtual int progress(double total_to_download, double downloaded);
+    virtual int mirror_failure(const char * msg, const char * url);
     virtual ~PackageTargetCB() = default;
 };
 
-/**
-* @class PackageTarget
-*
-* @brief Wraps librepo PackageTarget
-*/
+/// Wraps librepo PackageTarget
 struct PackageTarget {
 public:
-    /** Enum of supported checksum types.
-    * NOTE! This enum guarantee to be sorted by "hash quality"
-    */
+    // Enum of supported checksum types.
+    // NOTE! This enum guarantee to be sorted by "hash quality"
     enum class ChecksumType {
```

Comment:
Why we need it? please could you look at Package class, there is a similar enum.